### PR TITLE
refactor: evaluate all oneonone conversations protocol [WPB-5812]

### PR DESCRIPTION
**Pull request by markusfassbender**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-5812" title="WPB-5812" target="_blank">Sub-taskWPB-5812</a>  [iOS] Implement the logic for evaluating all 1:1 conversations
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

💡 merges in epic branch #944 

Recognised that we should use the protocol `SyncProgress` instead of `SyncStatus` class. Then we can use sorcery to build mocks and delete the custom ones.

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

- Run test suite.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by agisilaos — APPROVED** [review]

File: wire-ios-data-model/WireDataModel.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2738,7 +2722,6 @@
 			isa = PBXGroup;
 			children = (
 				597A3BB52B6418090020E337 /* Availability */,
-				597A3BB82B641EA20020E337 /* UserStatus */,
```
Comment (caldrian) [inline wire-ios-data-model/WireDataModel.xcodeproj/project.pbxproj:2741]: Why is this removed? Was it a duplicate reference?

**Review by caldrian — APPROVED** [review]

File: wire-ios-data-model/WireDataModel.xcodeproj/project.pbxproj (hunk 1 above)
Comment (markusfassbender) [reply wire-ios-data-model/WireDataModel.xcodeproj/project.pbxproj:2741]: Yes it was an empty lost reference in the project manager, probably because of a merge conflict.
